daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -446,9 +447,9 @@ private[lf] final class Compiler(
         )(
           env.toSEVar(choiceArgPos),
           env.toSEVar(cidPos),
-          translateExp(env, choice.controllers),
+          s.SEPreventCatch(translateExp(env, choice.controllers)),
           choice.choiceObservers match {
-            case Some(observers) => translateExp(env, observers)
+            case Some(observers) => s.SEPreventCatch(translateExp(env, observers))
             case None => s.SEValue.EmptyList
```

Comment:
We also need this change in `translateInterfaceChoiceBody`

Reply:
test added.